core/src/test/scala/cats/derived/adtdefns.scala

```diff
@@ -55,8 +55,19 @@ object TestDefns {
   final case class Recursive(i: Int, is: Option[Recursive])
   object Recursive extends ((Int, Option[Recursive]) => Recursive) {
 
-    implicit val arbitrary: Arbitrary[Recursive] =
-      Arbitrary(Gen.delay(Arbitrary.arbitrary[(Int, Option[Recursive])].map(tupled)))
+    implicit val arbitrary: Arbitrary[Recursive] = {
+      def tail(maxLength: Int): Gen[Option[Recursive]] =
+        if (maxLength <= 0) Gen.const(None)
+        else Gen.option(recursive(maxLength))
+
+      def recursive(maxLength: Int): Gen[Recursive] = for {
+        i <- Arbitrary.arbitrary[Int]
+        n <- Gen.choose(0, maxLength - 1)
+        is <- tail(n)
+      } yield Recursive(i, is)
```

Comment:
There's a good chance you only need one method for recursion.

Reply:
right :+1: